fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -145,6417 +145,6708 @@
-    private BigDecimal maxOutstandingLoanBalance;
-
-    @OneToMany(cascade = CascadeType.ALL, mappedBy = "loan", orphanRemoval = true, fetch=FetchType.LAZY)
-    @OrderBy(value = "expectedDisbursementDate, id")
-    private List<LoanDisbursementDetails> disbursementDetails = new ArrayList<>();
-
-    @OrderBy(value = "termApplicableFrom, id")
-    @OneToMany(cascade = CascadeType.ALL, mappedBy = "loan", orphanRemoval = true, fetch=FetchType.LAZY)
-    private List<LoanTermVariations> loanTermVariations = new ArrayList<>();
-
-    @Column(name = "total_recovered_derived", scale = 6, precision = 19)
-    private BigDecimal totalRecovered;
-
-    @OneToOne(cascade = CascadeType.ALL, mappedBy = "loan", optional = true, orphanRemoval = true, fetch=FetchType.EAGER)
-    private LoanInterestRecalculationDetails loanInterestRecalculationDetails;
-
-    @Column(name = "is_npa", nullable = false)
-    private boolean isNpa;
-
-    @Temporal(TemporalType.DATE)
-    @Column(name = "accrued_till")
-    private Date accruedTill;
-
-    @Column(name = "create_standing_instruction_at_disbursement", nullable = true)
-    private Boolean createStandingInstructionAtDisbursement;
-
-    @Column(name = "guarantee_amount_derived", scale = 6, precision = 19, nullable = true)
-    private BigDecimal guaranteeAmountDerived;
-
-    @Temporal(TemporalType.DATE)
-    @Column(name = "interest_recalcualated_on")
-    private Date interestRecalculatedOn;
-
-    @Column(name = "is_floating_interest_rate", nullable = true)
-    private Boolean isFloatingInterestRate;
-
-    @Column(name = "interest_rate_differential", scale = 6, precision = 19, nullable = true)
-    private BigDecimal interestRateDifferential;
-
-    @ManyToOne(fetch=FetchType.LAZY)
-    @JoinColumn(name = "writeoff_reason_cv_id", nullable = true)
-    private CodeValue writeOffReason;
-
-    @Column(name = "loan_sub_status_id", nullable = true)
-    private Integer loanSubStatus;
-
-    @Column(name = "is_topup", nullable = false)
-    private boolean isTopup = false;
-
-    @OneToOne(cascade = CascadeType.ALL, mappedBy = "loan", optional = true, orphanRemoval = true, fetch=FetchType.EAGER)
-    private LoanTopupDetails loanTopupDetails;
-
-    @OneToMany(fetch = FetchType.LAZY)
-    @JoinTable(name = "m_loan_rate", joinColumns = @JoinColumn(name = "loan_id"), inverseJoinColumns = @JoinColumn(name = "rate_id"))
-    private List<Rate> rates;
-
-    public static Loan newIndividualLoanApplication(final String accountNo, final Client client, final Integer loanType,
-            final LoanProduct loanProduct, final Fund fund, final Staff officer, final CodeValue loanPurpose,
-            final LoanTransactionProcessingStrategy transactionProcessingStrategy,
-            final LoanProductRelatedDetail loanRepaymentScheduleDetail, final Set<LoanCharge> loanCharges,
-            final Set<LoanCollateral> collateral, final BigDecimal fixedEmiAmount, final List<LoanDisbursementDetails> disbursementDetails,
```

Comment:
Please note, whenever we send a PR, it should contain either refactoring commit or new change commit. If the PR contains both then we should have first refactoring and then new feature commit.
I know its difficult, to judge initially whether the change you are going to make is going to be either refactoring or new change. But practicing such behavior makes it easy for reviewer. It also helps to understand the impact the PR is going to have on the existing functionality.